mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -1012,7 +1019,10 @@ getTensorExtractMemoryAccessPattern(tensor::ExtractOp extractOp,
   bool foundIndexOp = false;
   bool isContiguousLoad = isContiguousLoadIdx(linalgOp, extractOpTrailingIdx,
                                               foundIndexOp, resType);
-  isContiguousLoad &= foundIndexOp;
+  // TODO: Support generating contiguous loads for column vectors - that will
+  // require adding a permutation map to tranfer_read Ops.
+  bool isRowVector = resType.getShape().back() != 1;
+  isContiguousLoad &= (foundIndexOp && isRowVector);
```

Comment:
Is this logic actually being exercised by the second test case (the currently unsupported contiguous load) ? IIUC  isLoopInvariantIdx will return false for the second case and this line wont be needed?

Reply:
Are you referring to `@index_from_output_column_vector_contiguous_load`? 

This access is extracted from that test. The 1st idx is loop invariant (it's a constant) and the 2nd increments by 1:
```mlir
    %extracted = tensor.extract %src[%c0, %1] : tensor<8x128xf32>
```
And this is not a row vector:
```mlir
vector<8x1xf32>
```

So, without this condition (`... && isRowVector`), we'd be generated a contiguous load. Provided I didn't miss sth myself! :)

Reply:
You are right in that it increments by 1 but the current check in `isLoopInariantIdx` is 
`return linalgOp.getStaticLoopRanges()[indexOp.getDim()] == 1`
and isnt that `8 == 1` ?

Reply:
See for _which_ indices we call `isLoopInvariantIdx`: https://github.com/llvm/llvm-project/blob/1c47fa9b620d0abb280647b4f361ada43784d00e/mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp#L974-L991 

In particular, look at this line:
```cpp
  // The trailing index is not analysed here
  auto leadIndices = indices.drop_back(1);
```

The trailing index is analysed separately further down: https://github.com/llvm/llvm-project/blob/1c47fa9b620d0abb280647b4f361ada43784d00e/mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp#L1013

Reply:
Ah makes sense, thanks for expplaining.